bigquery/api/src/undelete_table.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * Copyright 2023 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * For instructions on how to run the full sample:
+ *
+ * @see https://github.com/GoogleCloudPlatform/php-docs-samples/tree/main/bigquery/api/README.md
+ */
+
+namespace Google\Cloud\Samples\BigQuery;
+
+# [START bigquery_undelete_table]
+use Google\Cloud\BigQuery\BigQueryClient;
+
+/**
+ * Restored a table from its snapshot for with deleted base table.
+ *
+ * @param string $projectId The project Id of your Google Cloud Project.
+ * @param string $datasetId The BigQuery dataset ID.
+ * @param string $snapshotId Snapshot ID of a snapshot with deleted base table
+ *  to be restored.
+ * @param string $restoredTableId TableId in which snapshot would be restored.
+ */
+function undelete_table(
+    string $projectId,
+    string $datasetId,
+    string $snapshotId,
+    string $restoredTableId
+): void {
+    $bigQuery = new BigQueryClient([
+        'projectId' => $projectId,
+    ]);
+    $dataset = $bigQuery->dataset($datasetId);
+    $snapshot = $dataset->table($snapshotId);
```

Comment:
Something like this?
```suggestion
    // Construct the restore-from table ID at current timestamp
    $snapshotEpoch = date_create()->format('Uv');
    $snapshotId = "{$tableId}@{$snapshotEpoch}";
    $snapshot = $dataset->table($snapshotId);
```

Reply:
Wow, that seems a really nice way to do this! I'll incorporate this.

Reply:
Thanks, applied in commit [8463095](https://github.com/GoogleCloudPlatform/php-docs-samples/pull/1774/commits/84630956b7a6f5c17a73325e954c6c186a19a898)